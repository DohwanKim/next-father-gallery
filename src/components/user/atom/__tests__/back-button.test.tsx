import { fireEvent, render } from '@testing-library/react';

import BackButton from '@/components/user/atom/back-button';

describe('BackButton', () => {
  it('BackButton 스냅샷 일치', () => {
    const { container } = render(<BackButton />);
    expect(container).toMatchSnapshot();
  });

  it('클릭하면 히스토리 백이 일어난다', () => {
    const back = jest.spyOn(window.history, 'back');
    const { getByRole } = render(<BackButton />);
    const button = getByRole('button');

    fireEvent.click(button);
    expect(back).toHaveBeenCalled();
  });
});
