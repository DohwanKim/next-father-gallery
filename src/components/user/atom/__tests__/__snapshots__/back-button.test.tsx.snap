// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton BackButton 스냅샷 일치 1`] = `
<div>
  <button
    class="flex items-center mb-5"
  >
    <svg
      class="lucide lucide-move-left mr-2"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 8L2 12L6 16"
      />
      <path
        d="M2 12H22"
      />
    </svg>
    <span>
      Back
    </span>
  </button>
</div>
`;
